sklearn/manifold/spectral_embedding_.py

```diff
@@ -287,12 +287,14 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
         # lobpcg needs double precision floats
         laplacian = check_array(laplacian, dtype=np.float64,
                                 accept_sparse=True)
-        laplacian = _set_diag(laplacian, 1, norm_laplacian)
+        laplacian = _set_diag(laplacian, 1 + 1e-5, norm_laplacian)
```

Comment:
This fix may be different from mine in https://github.com/scikit-learn/scikit-learn/pull/12316/commits/8cdcda2dfea2841bc2699b90219296020c481532 i.e.
```laplacian = laplacian + 1e-5 * sparse.eye(laplacian.shape[0]) ```
If `norm_laplacian = False` it has no effect, as far as I can see from the function `_set_diag`

May be add a comment here about why the diagonal shift is `1e-5` and how it affects the Laplacian eigenpairs.

Reply:
@lobpcg Thanks, I won't have time to look into this until tomorrow. Adding you as a collab in case you have a chance to verify `_set_diag` functionality and we need to make a change here.